src/dash-table/components/EdgeFactory.tsx

```diff
@@ -0,0 +1,228 @@
+import * as R from 'ramda';
+
+import { memoizeOne } from 'core/memoizer';
+
+import derivedDataEdges from 'dash-table/derived/edges/data';
+import derivedDataOpEdges from 'dash-table/derived/edges/operationOfData';
+import derivedFilterEdges from 'dash-table/derived/edges/filter';
+import derivedFilterOpEdges from 'dash-table/derived/edges/operationOfFilters';
+import derivedHeaderEdges from 'dash-table/derived/edges/header';
+import derivedHeaderOpEdges from 'dash-table/derived/edges/operationOfHeaders';
+import { EdgesMatrices, IEdgesMatrices } from 'dash-table/derived/edges/type';
+
+import getHeaderRows from 'dash-table/derived/header/headerRows';
+
+import { derivedRelevantCellStyles, derivedRelevantFilterStyles, derivedRelevantHeaderStyles } from 'dash-table/derived/style';
+import { Style, Cells, DataCells, BasicFilters, Headers } from 'dash-table/derived/style/props';
+
+import { ControlledTableProps, VisibleColumns, IViewportOffset, Data } from './Table/props';
+
+export default class EdgeFactory {
```

Comment:
This is the central change of this PR -- evaluates styles in isolation & reconciles styles for boundary cases (e.g. shared data / filter borders, etc.)